File: java/WesleySafadao.java
package br.com.umporcentovagabundo;

public class WesleySafadao {

    private final Integer dia;
    private final Integer mes;
    private final Integer ano;

    private WesleySafadao(Integer dia, Integer mes, Integer ano) {
        this.dia = dia;
        this.mes = mes;
        this.ano = ano;
    }

    private Integer getPercentualSafadeza() {
        Integer safadeza = 0;
        for (Integer atual = mes; atual > 0; atual--)
            safadeza += atual;
        return safadeza + (int) ((ano / 100f) * (50 - dia));
    }

    private Integer getPercentualAnjo() {
        return 100 - getPercentualSafadeza();
    }

    public void dizQualPercentualSafadezaEAnjo() {
        System.out.println(
            String.format(
                "Você é %1$s%% safado e %2$s%% anjo",
                getPercentualSafadeza(),
                getPercentualAnjo()));
    }

    public static WesleySafadao baseadoNaData(
                    Integer dia,
                    Integer mes,
                    Integer ano) {
        return new WesleySafadao(dia, mes, ano);
    }

}

File: fortran/WesleySafadao.f90
      ! FORTRAN IS ALIVE

      PROGRAM safadao
      
      CALL safadeza(8, 8, 90)

      END PROGRAM safadao

      SUBROUTINE safadeza(dia, mes, ano)
      
      INTEGER dia, mes, ano, soma_mes, i
      REAL*8 safado, anjo
      
      soma_mes = 0

      DO i = 1, mes
         soma_mes = soma_mes + i
      END DO

      safado = soma_mes + (ano / 100.) * (50 - dia)
      anjo = 100 - safado

      WRITE(*,'(a,F5.2,a,F5.2,a)')'Você é ',safado,'% safado e ',anjo,'% anjo'
      
      END SUBROUTINE safadeza    

